processFlows.json

```diff
@@ -1546,5 +1546,513 @@
+        "restriction": {
+          "MatchInputOutputMetadataValue": {
+            "inputIndex": 0,
+            "inputMetadataKey": "originalId",
+            "outputIndex": 2,
+            "outputMetadataKey": "demandA"
+          }
+        }
+      },
+      { "op": "And" },
+      {
+        "restriction": {
+          "MatchInputOutputMetadataValue": {
+            "inputIndex": 2,
+            "inputMetadataKey": "demandA",
+            "outputIndex": 2,
+            "outputMetadataKey": "demandA"
+          }
+        }
+      },
+      { "op": "And" },
+      {
+        "restriction": {
+          "MatchInputOutputMetadataValue": {
+            "inputIndex": 1,
+            "inputMetadataKey": "originalId",
+            "outputIndex": 2,
+            "outputMetadataKey": "demandB"
+          }
+        }
+      },
+      { "op": "And" },
+      {
+        "restriction": {
+          "MatchInputOutputMetadataValue": {
+            "inputIndex": 2,
+            "inputMetadataKey": "demandB",
+            "outputIndex": 2,
+            "outputMetadataKey": "demandB"
+          }
+        }
+      },
+      { "op": "And" },
+      {
+        "restriction": {
+          "MatchInputOutputMetadataValue": {
+            "inputIndex": 2,
+            "inputMetadataKey": "originalId",
+            "outputIndex": 2,
+            "outputMetadataKey": "originalId"
+          }
+        }
+      },
+      {
+        "restriction": {
+          "InputHasMetadata": {
+            "index": 2,
+            "metadataKey": "originalId"
+          }
+        }
+      },
+      {
+        "restriction": {
+          "None": {}
+        }
+      },
+      { "op": "NotL" },
+      {
+        "restriction": {
+          "MatchInputIdOutputMetadataValue": {
+            "inputIndex": 2,
+            "outputIndex": 2,
+            "outputMetadataKey": "originalId"
+          }
+        }
+      },
+      { "op": "And" },
+      { "op": "Or" },
+      { "op": "And" },
+      {
+        "restriction": {
+          "FixedInputMetadataValue": {
+            "index": 2,
+            "metadataKey": "state",
+            "metadataValue": {
+              "Literal": "acceptedFinal"
+            }
+          }
+        }
+      },
+      { "op": "And" },
+      {
+        "restriction": {
+          "FixedOutputMetadataValue": {
+            "index": 2,
+            "metadataKey": "state",
+            "metadataValue": {
+              "Literal": "cancelled"
+            }
+          }
+        }
+      },
+      { "op": "And" },
+      {
+        "restriction": {
+          "SenderHasOutputRole": {
+            "index": 2,
+            "roleKey": "MemberA"
+          }
+        }
+      },
+      {
+        "restriction": {
+          "SenderHasOutputRole": {
+            "index": 2,
+            "roleKey": "MemberB"
+          }
+        }
+      },
+      { "op": "Or" },
+      { "op": "And" },
+
+
+      { "op": "And" },
+      { "op": "Or" },
+      { "op": "And" }
```

Comment:
is the `comment` metadata file required or optional? If it's required can restrict with:
```
        "restriction": {
          "FixedOutputMetadataValueType": {
            "index": 2,
            "metadataKey": "comment",
            "metadataValueType": "File"
          }
        }
   ```

Reply:
I believe it's required. Thanks for pointing this out.